src/lsm/forest.zig

```diff
@@ -271,16 +272,23 @@ pub fn ForestType(comptime _Storage: type, comptime groove_cfg: anytype) type {
             };
 
             inline for (std.meta.fields(Grooves)) |groove_field| {
-                const groove = &@field(grooves, groove_field.name);
-                const Groove = @TypeOf(groove.*);
+                const Groove = std.meta.Child(groove_field.type);
                 const groove_options: Groove.Options = @field(grooves_options, groove_field.name);
 
-                groove.* = try Groove.init(allocator, node_pool, grid, groove_options);
+                @field(grooves, groove_field.name) = try Groove.init(
+                    allocator,
+                    node_pool,
+                    grid,
+                    groove_options,
+                );
                 grooves_initialized += 1;
             }
 
-            var compaction_pipeline =
-                try CompactionPipeline.init(allocator, grid, options.compaction_block_count);
+            const compaction_pipeline = try CompactionPipeline.init(
+                allocator,
+                grid,
+                options.compaction_block_count,
+            );
             errdefer compaction_pipeline.deinit(allocator);
 
             const scan_buffer_pool = try ScanBufferPool.init(allocator);
```

Comment:
Heh, incidently I was playing with copyhound on the weekend, and one thing I've noticed is that here and elsewhere we do

```
return Forest{
   ...
}
```

But, acccoding to the table at https://ziglang.org/documentation/master/#Result-Locations, that is going to be a stack-to-stack copy of the entire forest. 

If we do

```
return .{
   ...
};
```

then we'll get RLS, and the thing would be in-place constructed, so that'll actually prevent the copies. 

It _seems_ that the semantics we want here is to heap-allocate the entire replica, and then in-place construct the forest. 

Though, not sure if that's workable with error handling...

Reply:
Asked the hive mind here: https://ziggit.dev/t/result-location-semantics-and-error-handling/5767

Reply:
> Asked the hive mind here: https://ziggit.dev/t/result-location-semantics-and-error-handling/5767
:popcorn: 

I think we would need to reverse the logic, passing the result location to `init` in order to initialize it in place, like #2243.